failure when testing prereleases of dependencies

@rossbar We are getting the following failure when testing prereleases.  I suspect the issue is related to NumPy 1.22.0rc1.
```
=================================== FAILURES ===================================
________________________ TestLayout.test_spiral_layout _________________________

self = <networkx.drawing.tests.test_layout.TestLayout object at 0x7f4dd807dc70>

    def test_spiral_layout(self):
    
        G = self.Gs
    
        # a lower value of resolution should result in a more compact layout
        # intuitively, the total distance from the start and end nodes
        # via each node in between (transiting through each) will be less,
        # assuming rescaling does not occur on the computed node positions
        pos_standard = nx.spiral_layout(G, resolution=0.35)
        pos_tighter = nx.spiral_layout(G, resolution=0.34)
        distances = self.collect_node_distances(pos_standard)
        distances_tighter = self.collect_node_distances(pos_tighter)
>       assert sum(distances) > sum(distances_tighter)
E       assert 2.1859271805707814 > 2.4246843231297848
E        +  where 2.1859271805707814 = sum([0.31927215741496506, 0.3558859513351378, 0.4196367596771466, 0.5002552444945128, 0.5908770676490193])
E        +  and   2.4246843231297848 = sum([0.3585779234394243, 0.39751744460128174, 0.4657296446767588, 0.5524763132328394, 0.6503829971794806])

networkx/drawing/tests/test_layout.py:374: AssertionError
```

--- a/networkx/drawing/layout.py
+++ b/networkx/drawing/layout.py
@@ -1009,17 +1009,9 @@ def spiral_layout(G, scale=1, center=None, dim=2, resolution=0.35, equidistant=F
             pos.append([np.cos(theta) * r, np.sin(theta) * r])
 
     else:
-        # set the starting angle and step
-        step = 1
-        angle = 0.0
-        dist = 0.0
-        # set the radius for the spiral to the number of nodes in the graph
-        radius = len(G)
-
-        while dist * np.hypot(np.cos(angle), np.sin(angle)) < radius:
-            pos.append([dist * np.cos(angle), dist * np.sin(angle)])
-            dist += step
-            angle += resolution
+        dist = np.arange(len(G), dtype=float)
+        angle = resolution * dist
+        pos = np.transpose(dist * np.array([np.cos(angle), np.sin(angle)]))
 
     pos = rescale_layout(np.array(pos), scale=scale) + center
 

--- a/networkx/drawing/tests/test_layout.py
+++ b/networkx/drawing/tests/test_layout.py
@@ -15,17 +15,6 @@ class TestLayout:
         nx.add_path(cls.Gs, "abcdef")
         cls.bigG = nx.grid_2d_graph(25, 25)  # > 500 nodes for sparse
 
-    @staticmethod
-    def collect_node_distances(positions):
-        distances = []
-        prev_val = None
-        for k in positions:
-            if prev_val is not None:
-                diff = positions[k] - prev_val
-                distances.append((diff @ diff) ** 0.5)
-            prev_val = positions[k]
-        return distances
-
     def test_spring_fixed_without_pos(self):
         G = nx.path_graph(4)
         pytest.raises(ValueError, nx.spring_layout, G, fixed=[0])
@@ -367,20 +356,20 @@ class TestLayout:
         # intuitively, the total distance from the start and end nodes
         # via each node in between (transiting through each) will be less,
         # assuming rescaling does not occur on the computed node positions
-        pos_standard = nx.spiral_layout(G, resolution=0.35)
-        pos_tighter = nx.spiral_layout(G, resolution=0.34)
-        distances = self.collect_node_distances(pos_standard)
-        distances_tighter = self.collect_node_distances(pos_tighter)
+        pos_standard = np.array(list(nx.spiral_layout(G, resolution=0.35).values()))
+        pos_tighter = np.array(list(nx.spiral_layout(G, resolution=0.34).values()))
+        distances = np.linalg.norm(pos_standard[:-1] - pos_standard[1:], axis=1)
+        distances_tighter = np.linalg.norm(pos_tighter[:-1] - pos_tighter[1:], axis=1)
         assert sum(distances) > sum(distances_tighter)
 
         # return near-equidistant points after the first value if set to true
-        pos_equidistant = nx.spiral_layout(G, equidistant=True)
-        distances_equidistant = self.collect_node_distances(pos_equidistant)
-        for d in range(1, len(distances_equidistant) - 1):
-            # test similarity to two decimal places
-            assert distances_equidistant[d] == pytest.approx(
-                distances_equidistant[d + 1], abs=1e-2
-            )
+        pos_equidistant = np.array(list(nx.spiral_layout(G, equidistant=True).values()))
+        distances_equidistant = np.linalg.norm(
+            pos_equidistant[:-1] - pos_equidistant[1:], axis=1
+        )
+        assert np.allclose(
+            distances_equidistant[1:], distances_equidistant[-1], atol=0.01
+        )
 
     def test_rescale_layout_dict(self):
         G = nx.empty_graph()

